extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -30,7 +30,7 @@ export interface quickPickResults {
 const allowedFileExtensions: string[] = [FileExtension.Markdown, FileExtension.Notebook];
 
 export function hasSections(node: JupyterBookSection): boolean {
-	return node.sections !== undefined && node.sections.length > 0;
+	return node.sections !== undefined && node.sections?.length > 0;
```

Comment:
This shouldn't be needed - you already check that node.sections isn't undefined which the short-circuiting logic should mean that it never evaluates the second condition if that is undefined. Were you seeing differently?

Reply:
Yeah, I think the issue is that node.sections is null

Reply:
Ah gotcha - is this coming from some external library then? We shouldn't be using null ourselves but this makes sense if it's something we can't control.

Reply:
yes, it's from js-yaml when we parse the toc